### ISSUE TITLE
Automatically getting disconnected from neutrino node on startup

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#2278]
This is occurring off of master (79b3d959d3a6a2975489c570cd4b5d08d551618e) as well as `v0.4.0` release 

>2020-11-19T17:14:56UTC INFO [NeutrinoNode] Starting node
2020-11-19T17:14:56UTC INFO [DbValidate] Successfully validated 2 migrations (execution time 00:00.002s)
2020-11-19T17:14:56UTC INFO [DbMigrate] Current version of schema "main": 2
2020-11-19T17:14:56UTC INFO [DbMigrate] Schema "main" is up to date. No migration necessary.
2020-11-19T17:14:57UTC INFO [PeerMessageSender] Attempting to connect to peer=neutrino.suredbits.com/35.163.36.246:8333
2020-11-19T17:14:57UTC INFO [PeerMessageReceiver] Connection established with peer=Peer(neutrino.suredbits.com/35.163.36.246:8333)
2020-11-19T17:14:58UTC INFO [NeutrinoNode] Our peer=Peer(neutrino.suredbits.com/35.163.36.246:8333) has been initialized
2020-11-19T17:14:58UTC INFO [NeutrinoNode] Our node has been full started. It took=1744ms
2020-11-19T17:14:58UTC INFO [NeutrinoNode] Started node, best block hash 0000000000000000000065fa10f659bb758fbed07515a1190aab275174089366 at height 657680, with 656767 filter headers and 656767 filters
2020-11-19T17:14:59UTC INFO [DbValidate] Successfully validated 9 migrations (execution time 00:00.004s)
2020-11-19T17:14:59UTC INFO [DbMigrate] Current version of schema "main": 9
2020-11-19T17:14:59UTC INFO [DbMigrate] Schema "main" is up to date. No migration necessary.
2020-11-19T17:15:03UTC INFO [P2PClientActor] We've been disconnected by Peer(neutrino.suredbits.com/35.163.36.246:8333) command=PeerClosed
2020-11-19T17:15:06UTC INFO [BitcoinSServerMain] Starting neutrino node sync
2020-11-19T17:15:06UTC INFO [Server] Started Bitcoin-S HTTP server at /127.0.0.1:9999
2020-11-19T17:15:08UTC INFO [NeutrinoNode] Starting sync node, height=657680 hash=0000000000000000000065fa10f659bb758fbed07515a1190aab275174089366
2020-11-19T17:15:08UTC INFO [BitcoinSServerMain] Done starting Main!
2020-11-19T17:15:08UTC INFO [RepointableActorRef] Message [org.bitcoins.core.p2p.NetworkMessage$NetworkMessageImpl] to Actor[akka://bitcoin-s-server/user/P2PClient$-1605806097771#285507308] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://bitcoin-s-server/user/P2PClient$-1605806097771#285507308] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.

The key line is here

>2020-11-19T17:15:03UTC INFO [P2PClientActor] We've been disconnected by Peer(neutrino.suredbits.com/35.163.36.246:8333) command=PeerClosed

So basically right after server startup, we get disconnected.

**Comment by rorp** [comment]
It looks like bitcoind at `neutrino.suredbits.com` is not configured correctly.  In order for version 0.20.+ to serve filters theese two options should be set
```
blockfilterindex=1
peerblockfilters=1
```

**Comment by Christewart** [comment]
> It looks like bitcoind at `neutrino.suredbits.com` is not configured correctly. In order for version 0.20.+ to serve filters theese two options should be set
> 
> ```
> blockfilterindex=1
> peerblockfilters=1
> ```

The version is 

>bitcoind --version
Bitcoin Core Daemon version v0.18.99.0-3499e88
Copyright (C) 2009-2019 The Bitcoin Core developers

on our neutrino node

**Comment by rorp** [comment]
I can't reproduce it. 

Apparently, the bitcoin-s node sends an invalid P2P command (maybe with non-existing block/filter hash or height) to the Bitcoin Core node, and the Bitcoin Core node closes the connection.

**Comment by Christewart** [comment]
So with some debugging with @nkohen , i believe the problem might be with bitcoind. I think the number of connection count slots are full

We had 124 connections on our neutrino bitcoind node
>  bitcoin-cli getconnectioncount
124

after restarting the node to reset the connection count me and nadav didn't have problems connecting.
